barretenberg/cpp/src/barretenberg/honk/flavor/goblin_translator.hpp

```diff
@@ -0,0 +1,1630 @@
+        ACCUMULATOR_LOW_LIMBS_RANGE_CONSTRAINT_2_SHIFT,
+        ACCUMULATOR_LOW_LIMBS_RANGE_CONSTRAINT_3_SHIFT,
+        ACCUMULATOR_LOW_LIMBS_RANGE_CONSTRAINT_4_SHIFT,
+        ACCUMULATOR_LOW_LIMBS_RANGE_CONSTRAINT_TAIL_SHIFT,
+        ACCUMULATOR_HIGH_LIMBS_RANGE_CONSTRAINT_0_SHIFT,
+        ACCUMULATOR_HIGH_LIMBS_RANGE_CONSTRAINT_1_SHIFT,
+        ACCUMULATOR_HIGH_LIMBS_RANGE_CONSTRAINT_2_SHIFT,
+        ACCUMULATOR_HIGH_LIMBS_RANGE_CONSTRAINT_3_SHIFT,
+        ACCUMULATOR_HIGH_LIMBS_RANGE_CONSTRAINT_4_SHIFT,
+        ACCUMULATOR_HIGH_LIMBS_RANGE_CONSTRAINT_TAIL_SHIFT,
+        QUOTIENT_LOW_BINARY_LIMBS_SHIFT,
+        QUOTIENT_HIGH_BINARY_LIMBS_SHIFT,
+        QUOTIENT_LOW_LIMBS_RANGE_CONSTRAINT_0_SHIFT,
+        QUOTIENT_LOW_LIMBS_RANGE_CONSTRAINT_1_SHIFT,
+        QUOTIENT_LOW_LIMBS_RANGE_CONSTRAINT_2_SHIFT,
+        QUOTIENT_LOW_LIMBS_RANGE_CONSTRAINT_3_SHIFT,
+        QUOTIENT_LOW_LIMBS_RANGE_CONSTRAINT_4_SHIFT,
+        QUOTIENT_LOW_LIMBS_RANGE_CONSTRAINT_TAIL_SHIFT,
+        QUOTIENT_HIGH_LIMBS_RANGE_CONSTRAINT_0_SHIFT,
+        QUOTIENT_HIGH_LIMBS_RANGE_CONSTRAINT_1_SHIFT,
+        QUOTIENT_HIGH_LIMBS_RANGE_CONSTRAINT_2_SHIFT,
+        QUOTIENT_HIGH_LIMBS_RANGE_CONSTRAINT_3_SHIFT,
+        QUOTIENT_HIGH_LIMBS_RANGE_CONSTRAINT_4_SHIFT,
+        QUOTIENT_HIGH_LIMBS_RANGE_CONSTRAINT_TAIL_SHIFT,
+        RELATION_WIDE_LIMBS_SHIFT,
+        RELATION_WIDE_LIMBS_RANGE_CONSTRAINT_0_SHIFT,
+        RELATION_WIDE_LIMBS_RANGE_CONSTRAINT_1_SHIFT,
+        RELATION_WIDE_LIMBS_RANGE_CONSTRAINT_2_SHIFT,
+        RELATION_WIDE_LIMBS_RANGE_CONSTRAINT_3_SHIFT,
+        ORDERED_RANGE_CONSTRAINTS_0_SHIFT,
+        ORDERED_RANGE_CONSTRAINTS_1_SHIFT,
+        ORDERED_RANGE_CONSTRAINTS_2_SHIFT,
+        ORDERED_RANGE_CONSTRAINTS_3_SHIFT,
+        ORDERED_RANGE_CONSTRAINTS_4_SHIFT,
+
+        Z_PERM_SHIFT,
+        /*All precomputed polynomials*/
+        LAGRANGE_FIRST,
+        LAGRANGE_LAST,
+        LAGRANGE_ODD,
+        LAGRANGE_EVEN,
+        LAGRANGE_SECOND,
+        LAGRANGE_SECOND_TO_LAST_IN_MINICIRCUIT,
+        ORDERED_EXTRA_RANGE_CONSTRAINTS_NUMERATOR,
+        /*Utility value*/
+        TOTAL_COUNT
+
+    };
+
+    using CircuitBuilder = GoblinTranslatorCircuitBuilder;
+    using PCS = pcs::kzg::KZG<curve::BN254>;
+    using Curve = curve::BN254;
+    using GroupElement = Curve::Element;
+    using Commitment = Curve::AffineElement;
+    using CommitmentHandle = Curve::AffineElement;
+    using CommitmentKey = pcs::CommitmentKey<Curve>;
+    using VerifierCommitmentKey = pcs::VerifierCommitmentKey<Curve>;
+    using FF = Curve::ScalarField;
+    using BF = Curve::BaseField;
+    using Polynomial = barretenberg::Polynomial<FF>;
+    using PolynomialHandle = std::span<FF>;
+
+    // The size of the circuit which is filled with non-zero values for most polynomials. Most relations (everything
+    // except for Permutation and GenPermSort) can be evaluated just on the first chunk
+    static constexpr size_t MINI_CIRCUIT_SIZE = mini_circuit_size;
```

Comment:
where does mini_circuit_size come from? It would be clearer if this param specifies a class or namespace if it's located in another file

Reply:
It is the template parameter. We instantiate with 2048 at the end of file

Reply:
Since this is a VM this is supposed to be fixed, so I decided to use a parameter for simpler concatenation